game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java

```diff
@@ -85,9 +85,11 @@ public void setChat(final Chat chat) {
    */
   private void setDynamicPreferredSize() {
     int maxNameLength = 0;
-    final FontMetrics fontMetrics = this.getFontMetrics(UIManager.getFont("TextField.font"));
-    for (final UserName onlinePlayer : chat.getOnlinePlayers()) {
-      maxNameLength = Math.max(maxNameLength, fontMetrics.stringWidth(onlinePlayer.getValue()));
+    if (chat != null) {
```

Comment:
I was thinking about this for a while, but I think this isn't the right fix for this.
The stacktrace in #7034 makes it clear that this `NullPointerException` occurs when shutting down a game, so adjusting the preffered size just before the game is closed doesn't really make sense to me.

That's why I'd recommend making this method no-op if `chat` is null i.e.
```java
if (chat == null) {
  return;
}
```
At the start if the method

Reply:
Just pushed that, and verifying zero interactions with renderer mock. Was wondering if I should create more unit tests/test cases (in general) ?